### DISCUSSION TITLE
Please Do Not Remove the MapboxLayer API from v9

### DISCUSSION THREAD

**Discussion by geostarters** [visgl/deck.gl#8700] [Q&A] [answered]
We kindly request that you do not remove the MapboxLayer API in version 9. This functionality allows us, for example, with MapLibre, to integrate a 3D tiles model of cities as an additional layer. When using addLayer, we can control the rendering order of the layer and utilize the "symbols" layers already defined in the (mapbox) style, enabling us, for instance, to label the points of interest (POIs).

![bcn_labels](https://github.com/visgl/deck.gl/assets/7668442/94812cc6-1591-46c5-be19-8fbadf48928c)

While we understand that similar functionality can be achieved with MapBoxOverlay as a Control and then  adding an MVT layer as annotation, we believe that MapboxLayer and MapboxOverlay should not be mutually exclusive. Both are utilized depending on the specific use case.

Thank you for considering our request.

**Comment by Pessimistress** [comment, marked as answer]
What you are doing can be achieved with MapboxOverlay with `interleaved: true`. The layer will be inserted between maplibre layers if `beforeId` is specified.

**Reply by geostarters** [reply]
Thanks for the answer. we will try it.
